core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1883,9 +1883,16 @@ private static SslContext findDefaultSslContext(VirtualHost defaultVirtualHost,
 
     private static void warnIfServiceHasMultipleRoutes(String path, HttpService service) {
         if (service instanceof ServiceWithRoutes) {
+            if (!Flags.reportIgnoredMultipleRoutes()) {
+                return;
+            }
+
             if (((ServiceWithRoutes) service).routes().size() > 0) {
                 logger.warn("The service has self-defined routes but the routes will be ignored. " +
-                            "It will be served at the route you specified: path={}, service={}",
+                            "It will be served at the route you specified: path={}, service={}. " +
+                            "If this is an intended behavior (e.g. The path is a prefix and the service " +
+                            "still works under the prefix) you can disable this log message by specifying " +
```

Comment:
nit; I find "works" to be ambiguous.  OK with the original version though.

```suggestion
                            "If this is intended behavior (i.e. The service is served " +
                            "under the prefix) you can disable this log message by passing " +
```

Reply:
Thanks for the rephrasing. I removed `i.e ...` part though because I added a method that adds the prefix to the routes. 😄